### PR TITLE
First attempt at using Konnectivity

### PR DESCRIPTION
**Pull request by mreiger**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -387,14 +391,21 @@ func checkService(opts *Opts, client *k8s.Clientset) error {
...
 }
 
-func runForwarder(serviceIP string, servicePort int, opts *Opts) {
+func runForwarder(serviceIP, servicePort string, opts *Opts) {
 	for {
 		logger.Info("Starting forwarder")
+		fluentTargetIP := serviceIP
+
+		if opts.KonnectivityUDSSocket != "" {
+			// TODO set up the tunnel
```
Comment (Gerrit91) [inline main.go]: There is a TODO?

Comment (mreiger) [reply main.go]: Oops, that was a leftover old comment. TODO is now done. :smile:

File: main.go [hunk 2]
```diff
@@ -387,14 +391,22 @@ func checkService(opts *Opts, client *k8s.Clientset) error {
...
 
-func runForwarder(serviceIP string, servicePort int, opts *Opts) {
+func runForwarder(serviceIP, servicePort string, opts *Opts) {
 	for {
 		logger.Info("Starting forwarder")
+		fluentTargetIP := serviceIP
+
+		if opts.KonnectivityUDSSocket != "" {
+			ctx, cancel := context.WithCancel(context.Background())
+			go konnectivityproxy.MakeProxy(ctx, opts.KonnectivityUDSSocket, serviceIP, servicePort, logger)
```
Comment (Gerrit91) [inline main.go]: If opening the proxy fails with `Could not open port for listening`, does the controller shut down or will it just continue to run without being able to forward messages?

**Review by Gerrit91 — APPROVED** [review]
I think I am fine with these changes. Did you take code for the proxy from another project? Maybe can you add a reference or use it as a dependency? 

The asynchronous nature of the code makes it hard for me to determine if the whole machinery can get stuck in limbo under certain failure conditions, so I left one comment out of curiosity.

File: main.go (hunk 2 above)
Comment (mreiger) [reply main.go]: You were right; in this case the controller would get stuck.

I rewrote the proxy module with (I hope) a cleaner interface that would handle this case too, in a new branch. PR: https://github.com/metal-stack/audit-forwarder/pull/4

(I also added a reference to the source of the proxying code there.)

**Comment by mreiger** [comment]
Closing because superseded by https://github.com/metal-stack/audit-forwarder/pull/4

**Pull request closed without merging**